pkg/resource/notebook_instance/hooks.go

```diff
@@ -48,3 +52,33 @@ func (rm *resourceManager) customSetOutput(r *resource) {
 	latestStatus := r.ko.Status.NotebookInstanceStatus
 	svccommon.SetSyncedCondition(r, latestStatus, &resourceName, &modifyingStatuses)
 }
+
+// CustomSetOutputDescribe has two functions
+// 1. Setting the resource synced condition during sdk.Find
+// 2. Starting the Notebook if the StoppedByController field is set to UpdateTriggerd
+// 3. Reseting the StoppedByController field.
+func (rm *resourceManager) customSetOutputDescribe(r *resource,
+	ko *svcapitypes.NotebookInstance) error {
+	notebookStatus := *ko.Status.NotebookInstanceStatus // Get the Notebook State
+	if notebookStatus == svcsdk.NotebookInstanceStatusStopped {
+		if ko.Status.StoppedByControllerMETA != nil && *ko.Status.StoppedByControllerMETA == "UpdateTriggered" {
+			err := rm.startNotebookInstance(r)
+			if err != nil {
+				//We dont update StoppedByController here because we want it to try to start it again on the next reqeue.
```

Comment:
Have you been able to test that this is working as expected ? Do you have a negative test case for updates ?

Reply:
nit: use correct variable name in comments too.

Reply:
It will be tested in the unit tests.  startNotebookInstance rarely returns errors.